fe/fe-core/src/main/java/com/starrocks/qe/SetExecutor.java

```diff
@@ -54,13 +54,25 @@ private void setVariable(SetVar var) throws DdlException {
             // do nothing
             return;
         } else {
-            VariableMgr.setVar(ctx.getSessionVariable(), var);
+            VariableMgr.setVar(ctx.getSessionVariable(), var, false);
         }
     }
 
-    public void execute() throws DdlException {
+    private boolean isSessionVar(SetVar var) {
+        return !(var instanceof SetPassVar
+                || var instanceof SetNamesVar
+                || var instanceof SetTransaction);
+    }
+
+    public void execute(boolean onlySessionVar) throws DdlException {
         for (SetVar var : stmt.getSetVars()) {
-            setVariable(var);
+            if (onlySessionVar) {
```

Comment:
What's the difference between VariableMgr.setVar and setVariable?

Reply:
VariableMgr.setVar will only set the session variable.
setVariable will set the session variables and password. 
```java
private void setVariable(SetVar var) throws DdlException {
        if (var instanceof SetPassVar) {
            // Set password
            SetPassVar setPassVar = (SetPassVar) var;
            ctx.getCatalog().getAuth().setPassword(setPassVar);
        } else if (var instanceof SetNamesVar) {
            // do nothing
            return;
        } else if (var instanceof SetTransaction) {
            // do nothing
            return;
        } else {
            VariableMgr.setVar(ctx.getSessionVariable(), var, false);
        }
}
```